test/unit/model/jdk-install-test.js

```diff
@@ -162,7 +162,26 @@ describe('JDK installer', function() {
       });
     });
 
+    it('should create deafult empty callback if not provided', function() {
+      let jdk = new JdkInstall(installerDataSvc, 'url', 'file');
+      mockDetectedJvm('1.8.0_1');
+      try {
+        jdk.detectExistingInstall();
+      } catch (exception)  {
+        expect.fail('Did not created default empty callback');
+      }
+    });
 
+    it('should check for available msi installtion', function(done) {
+      mockDetectedJvm('1.8.0_1');
+      let jdk = new JdkInstall(installerDataSvc, 'url', 'file');
+      jdk.findMsiInstalledJava.restore();
+      return jdk.detectExistingInstall(function() {
+        expect(executeFileStub).to.have.been.calledOnce;
+        expect(writeFileStub).to.have.been.calledOnce;
```

Comment:
I have updated updated PR. Now it checks if windows specific code is called and worked as expected, else branch does the opposite and verifies windows specific code is not called on other platforms.